### ISSUE TITLE
How to add this to a 1.12.2 forge project

### ISSUE THREAD

**Issue by Doogie13** [cabaletta/baritone#3641]
## What do you need help with?
Is there an actual guide on adding baritone to a mod? If so, please link it, else, could someone tell me how I would go about doing so?
Much appreciated.

## Final checklist
- [x] I know how to properly use check boxes 
- [ ] I have not used any OwO's or UwU's in this issue.

**Comment by ZacSharp** [comment]
No, there's no guide on how to do it and "just add it to your libraries" usually ends in trouble if you want the `runClient` to work and is difficult (but much easier) if you only want to compile and test in a normal minecraft install.
If you want an example on how to do it that way look at mods like [lambda](https://github.com/lambda-client/lambda) (don't know whether they have a working `runClient` task).

If you want to avoid that trouble you can also compile Baritone with a slightly tweaked buildscript so it uses srg names so you can dump that special build into your libs folder and it works like any other mod.

**Comment by Doogie13** [comment]
> If you want to avoid that trouble you can also compile Baritone with a slightly tweaked buildscript so it uses srg names so you can dump that special build into your libs folder and it works like any other mod.

This sounds promising, would I just have to remove the part where it reobfuscates the jar? I don't want this mod to bundle baritone, so this seems like it would work?

**Comment by Doogie13** [comment]
I have found the solution to having Baritone intergration in my mod without forcing it to be present. 
`maven { url = 'https://impactdevelopment.github.io/maven/' }`
`implementation "cabaletta:baritone-api:1.2"`

I still need assistance building baritone and preventing it from reobfuscating itself so I can use that jar in my dev environment mods folder

**Comment by ZacSharp** [comment]
<details><summary>Excerpt from the readme of my own project</summary><pre>This mod needs a srg mapped Baritone build to compile.


To get a srg mapped build get the Baritone source code from https://github.com/cabaletta/baritone and add

reobf.jar.mappingType = 'SEARGE'

to the end of build.gradle to change the mappings, then run a normal build.


For debugging purposes a unoptimized Baritone build can be helpful. To get one you can copy the `baritone` directory from the baritone-unoptimized jar into any of the baritone-forge jars of the same version.


To compile this project against your Baritone version place the file in `libs` and build the project using gradle.
</pre></details>
<details><summary>Excerpt from the buildscript</summary><pre>runClient {
    args "--tweakClass","org.spongepowered.asm.launch.MixinTweaker"
    jvmArgs "-enableassertions"
}
</pre></details>

Since the version of ForgeGradle I'm using (an ancient one) automatically includes jars from `${PROJECT_ROOT}/lib` I only had to add the `--tweakClass` argument to make mixin work and Baritone was working both for compilation and testing. (the `-enableassertions` is so e.g.
```java
assert state.isValid();
```
actually throws an error if the state is not valid.

**Comment by Doogie13** [comment]
one last thing, trying to get proper sources of baritone for use inside my project using 
```gradle
implementation 'com.github.cabaletta:baritone:1.2.15'
``` 
has not worked. I don't know exactly where it is supposed to get this jar from but it would allow me to use sources rather than the jar which is meant to be bundled since that breaks a ton of stuff and has no javadoc.

**Comment by ZacSharp** [comment]
Sorry, I can't help you with that (I just use one of my Baritone workspaces whenever I need the sources/javadoc).
To which buildscript did you add the `reobf.jar.mappingType = 'SEARGE'`?

**Comment by Doogie13** [comment]
I added that as the bottom line of build.gradle which now that I write it out is retarded since iirc there is a buildscript block. Will have to get back to you

**Comment by ZacSharp** [comment]
No, last line of the file is correct (actually almost any line is fine, but last is a safe and easy location). I asked because you have to add it to *Baritones* build.gradle and build *Baritone* with it so you can copy the resulting baritone-api-forge-whatever.jar over to *your* project to build *your* project with it. The second change (`runClient {args "--tweakClass","org.spongepowered.asm.launch.MixinTweaker"}`) is then applied to *your* build.gradle to correctly load Baritone when using runClient. Don't try getting an unoptimized Forge build before you have the rest working.
